core/transformations.h

```diff
@@ -1914,6 +1904,87 @@ void radialAverageAxis(const MultidimArray< T >& in, char axis, MultidimArray< d
 		REPORT_ERROR(ERR_ARG_INCORRECT,"Not implemented yet");
 }
 
+template<typename T>
+void radialAverageNonCubic(const MultidimArray< T >& m,
+                   Matrix1D< int >& center_of_rot,
+                   MultidimArray< T >& radial_mean,
+                   MultidimArray< int >& radial_count,
+                   const bool& rounding = false)
+{
+    Matrix1D< double > idx(3);
+
+    size_t sizemax = std::max({XSIZE(m), YSIZE(m), ZSIZE(m)});
+    double scalex = XSIZE(m)/sizemax;
+    double scaley = YSIZE(m)/sizemax;
+    double scalez = ZSIZE(m)/sizemax;
+
+    // If center_of_rot was written for 2D image
+    if (center_of_rot.size() < 3)
+        center_of_rot.resize(3);
+
+    // First determine the maximum distance that one should expect, to set the
+    // dimension of the radial average vector
+    MultidimArray< int > distances(8);
+
+    double z = scalez * (STARTINGZ(m) - ZZ(center_of_rot));
+    double y = scaley * (STARTINGY(m) - YY(center_of_rot));
+    double x = scalex * (STARTINGX(m) - XX(center_of_rot));
+
+    distances(0) = (int) floor(sqrt(x * x + y * y + z * z));
```

Comment:
please do the similar change as here https://github.com/I2PC/xmippCore/pull/106/files#diff-732e50819a85d2592075588477a1631f594fc66f719692cfd6a075960091ccc2R1811

Reply:
I've made a mistake, the code in please do the similar change as here https://github.com/I2PC/xmippCore/pull/106/files#diff-732e50819a85d2592075588477a1631f594fc66f719692cfd6a075960091ccc2R1811 should be actually here. I didn't want to modify precomputeDistance function as I am not using it. When I did that optimization I was thinking that I was modifying radaverageNonCubic.